### PR TITLE
Update DOM so that <form> has various 'on*' props

### LINKED ISSUE #45: Form doesn't support onSubmit
According to [React documentation](https://reactjs.org/docs/events.html#form-events), `form` elements should support the following events:
- `onChange`
- `onInput`
- `onInvalid`
- `onSubmit`

However, they don't seem to be accepted as props in the [definition of `form`](https://github.com/lumihq/purescript-react-basic/blob/master/src/React/Basic/DOM.purs#L679)

### PR DESCRIPTION
**Pull request by zudov**
Fixes #45

### CODE REVIEW

File: src/React/Basic/DOM.purs [hunk 1]
```diff
@@ -682,6 +683,10 @@ type Props_form =
   , children :: Array JSX
   , method :: String
   , name :: String
+  , onChange :: EventHandler
+  , onInput :: String
+  , onInvalid :: String
+  , onSubmit :: String
```
Comment (f-f) [inline src/React/Basic/DOM.purs]: Shouldn't this be an `EventHandler`? (and I think also the other ones?)

Comment (justinwoo) [reply src/React/Basic/DOM.purs]: time to PR upstream again probably :100:

Comment (f-f) [reply src/React/Basic/DOM.purs]: Nah, in the upstream there was no place to define these things AFAIK

File: codegen/index.js [hunk 2]
```diff
@@ -20,6 +20,10 @@ const types = {
   "min": "Number",
   "multiple": "Boolean",
   "muted": "Boolean",
+	"onClick": "EventHandler",
+	"onInput": "EventHandler",
+	"onInvalid": "EventHandler",
+	"onSubmit": "EventHandler",
```
Comment (f-f) [inline codegen/index.js]: Not sure if Github is displaying them in a weird way or they are not aligned properly?

Comment (justinwoo) [reply codegen/index.js]: wtf tabs

Comment (zudov) [reply codegen/index.js]: omg, somehow they slipped in

**Comment by zudov** [comment]
Simply rerunning codegen with `react-html-attributes` resulted in these changes (added `on*` properties to `form`). It doesn't seem like they changed anything that would require changes in codegen, only the missing properties were added.

**Comment by spicydonuts** [comment]
Oh right, sorry. Looks good.

**Pull request merged**